Update the CIS AWS guide for version 1.3.0 (#392)
gruntwork-io/gruntwork-io.github.io: _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

This is a complete guide to help you achieve compliance with the
link:https://www.cisecurity.org/benchmark/amazon_web_services/[CIS AWS Foundations Benchmark]. By following this guide,
you can launch infrastructure that is compliant with the Benchmark recommendations, and you'll be set to retain a
compliant state over time because all of the infrastructure is defined as code. This guide targets version 1.3.0 of the Benchmark.

> *Note:* Previously, we supported version 1.2.0 of the Benchmark. If you are looking to upgrade from v1.2.0 to v1.3.0,

Author: Fixed in https://github.com/gruntwork-io/gruntwork-io.github.io/pull/392/commits/5cd3f1c19af4b37e0f4ecd1980e2da0f1e7b2b00.

[[core_concepts]]
== Core concepts

> The CIS AWS Foundations Benchmark is organized into five sections:

Reviewer: Maybe make this future-proof by saying:
```suggestion
The CIS AWS Foundations Benchmark is organized into the following sections:
```

 +
 +

. Identity and Access Management

> . Storage

Reviewer: Suggestion for ease of navigation: Is it worth making everything in this bullet list a link to the section below? That said, I'm not sure which makes most sense. 

One option:
Right here, link to the `Recommendation Sections` which has a summary:

- ` - link:https://github.com/gruntwork-io/gruntwork-io.github.io/blob/ca410b41956807bc9f2cb1b71c9fbe2f839eb1ef/_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc#identity-and-access-management[Identity and Access Management]`

Then in that section ^ (under Production-grade Design), make the header itself a link to the main section:
- `link:https://github.com/gruntwork-io/gruntwork-io.github.io/blob/ca410b41956807bc9f2cb1b71c9fbe2f839eb1ef/_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc#identity-and-access-management-1[Identity and Access Management]`

Everything is linked at the TOC level, but you have to scroll to the top of the page to get there.

Reviewer: Or, just link to the main section from here?

Reviewer 2: @rhoboat hi Rho sorry for the delayed reply! I think I understand your point, but because this guide is directly consumed on the [Gruntwork marketing site](https://deploy-preview-392--keen-clarke-470db9.netlify.app/guides/compliance/how-to-achieve-cis-benchmark-compliance/#what-youll-learn-in-this-guide), does that left side-bar actually satisfy your point? As in I'm typically using that myself vs scrolling to the top of the page.

Reviewer 2: hi @rhoboat gentle bump to resolve this one as we approach the finish line!

Reviewer 3: Hey, so just confirming - is this something that we can update later on? I guess the impact would be small, if we just published the guide as is - and come back with the CIS Service Catalog work to re-do a few things (we could bring back up the discussion then)

Reviewer 2: Yes, I don't feel it's blocking. Let's resolve it for now and revisit this issue later if need be.

[source,json]
----
{
	"ServerCertificateMetadataList": [{
		"ServerCertificateId": "EHDGFRW7EJFYTE88D",
		"ServerCertificateName": "MyServerCertificate",
		"Expiration": "2020-07-10T23:59:59Z",
		"Path": "/",
		"Arn": "arn:aws:iam::012345678910:server-
		certificate / MySSLCertificate ",
		"UploadDate": "2018-06-10T11:56:08Z"
	}]
}
----

The `Expiration` attribute contains the expiration date for each SSL/TLS certificate which you can use to determine
if it should be removed. To remove the certificate use the `delete-server-certificate` command, making sure to
substitute `<CERTIFICATE_NAME>` with the `ServerCertificateId` attribute from the previous command:

[source,bash]
----
aws iam delete-server-certificate --server-certificate-name <CERTIFICATE_NAME>
----

To automate this process you might decide to implement a Lambda function that runs on a regular schedule and removes
all expired SSL/TLS certificates. Check out the link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/examples/cleanup-expired-certs/terraform[example]
provided in the Gruntwork Service Catalog to see how to deploy a ready-made module that can be

> deployed in each of your AWS accounts.

Author: Fixed in https://github.com/gruntwork-io/gruntwork-io.github.io/pull/392/commits/5cd3f1c19af4b37e0f4ecd1980e2da0f1e7b2b00.

[[iam_access_analyzer]]
==== IAM Access Analyzer

> The latest CIS guide requires that the AWS IAM Access Analyzer service is enabled across all active regions in a given

Reviewer 2: ```suggestion
The new version 1.3.0 recommendations stipulate that the AWS IAM Access Analyzer service is enabled across all active regions in a given
```

After re-reading this whole document, I think we explicitly want to mention CIS v1.3.0 here as the wording in 'latest CIS guide' will quickly become out of date.

AWS Account or Organization.

To achieve this compliance requirement, enable the IAM Access Analyzer service for every AWS region you have enabled in
your account. Alternatively, you could make use of the

> link:https://github.com/gruntwork-io/terraform-aws-security/blob/master/modules/iam-access-analyzer-multi-region[module]

Author: Fixed in https://github.com/gruntwork-io/gruntwork-io.github.io/pull/392/commits/5cd3f1c19af4b37e0f4ecd1980e2da0f1e7b2b00.

security groups, and remote access. Moreover, our link:https://gruntwork.io/reference-architecture/[Reference
Architecture] can get you up and running with a secure network configuration immediately.

> Recommendation 5.1 forbits allowing access to the remote server administration ports, such as SSH to port 22 and Remote

Reviewer 3: ```suggestion
Recommendation 5.1 forbids allowing access to the remote server administration ports, such as SSH to port 22 and Remote
```

Desktop to port 3389, from the public Internet (`0.0.0.0/0`) on the level of Network ACL rules. Recommendation 5.2 does

> the same for Security Groups.  This seems entirely reasonable, no? Avoid creating any Security Groups and NACL with

Reviewer 3: Is this intentionally left here? 😄 
```suggestion
the same for Security Groups.  Avoid creating any Security Groups and NACL with
```

Author: This was there in the previous version of the guide, and the context around these recommendations hasn't changed, so I saw no need to reformulate the accompanying text. Happy to incorporate any specific suggestions though!

Reviewer 3: ahh okay! Happy to leave it as it was - I raised it because it seemed like the tone of that specific sentence was a bit different to everything else in this section :)

TODO(Rob): Add instructions for 1.19.

[[iam_access_analyzer_deployment]]
=== Steps to enable IAM Access Analyzer for all enabled AWS regions
To enable IAM Access Analyzer in your AWS Account or Organization, you need to do it separately for every region.

To enable this for your AWS account:
* you can do so manually using the AWS CLI or the AWS web console. For more guidance on how to do so, please refer to
link:https://docs.aws.amazon.com/IAM/latest/UserGuide/what-is-access-analyzer.html[the official AWS docs];

> * you can also use the examples provided as part of `learning-and-testing/landingzone`:

Author: I think here in the deployment section we should specifically instruct to use our IaC library as opposed to manual configuration. So you can remove the bit that talks about manual enablement of this feature.

[[cleanup_expired_certs_deployment]]
=== Automatically remove expired SSL/TLS certificates from IAM
Next, use the link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/blob/master/modules/cleanup-expired-certs/README.adoc[`cleanup-expired-certs`]
module to deploy a Lambda function to remove all expired SSL/TLS certificates from AWS IAM. This module will add all of the relevant IAM resources needed to
periodically execute the Lambda function and optionally, report metrics to CloudWatch. Be sure to deploy this module into each of your AWS accounts.

[[iam_access_analyzer_deployment]]
=== Steps to enable IAM Access Analyzer for all enabled AWS regions
To enable IAM Access Analyzer in your AWS Account or Organization, you need to do it separately for every region.

To enable this for your AWS account:
* use the examples provided as part of `learning-and-testing/landingzone`:
link:https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-learning-and-testing/landingzone/account-baseline-root[account-baseline-root]

> and link:https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-learning-and-testing/landingzone/account-baseline-security[account-baseline-security].

Reviewer 2: @brikis98 @ina-stoyanova I know above and below we've spoken about using Terragrunt examples here in the deployment guide, but I can see you've linked to a Landing Zone example in our service catalog. Also below I noticed you're inserting the actual Terraform module definition. Would it make more sense to rework this like I've done in the Cleanup Expired Certs section and push users down the route of only deploying one module or should we be more clear about requiring users to deploy our account baselines? For example, you could change this section to something along the lines of:

```
If your using Terragrunt, then you can invoke the standalone module directly by following the pattern outlined in
<<wrapper_code_terragrunt>> and using a `terragrunt.hcl` file with the following contents:

.infrastructure-live/<your account>/_global/iam-access-analyzer/terragrunt.hcl
[source,hcl]
----
terraform {
  # Make sure to replace <VERSION> in this URL with the latest terraform-aws-security release
  source = "git::ssh://git@github.com/gruntwork-io/terraform-aws-security.git//modules/iam-access-analyzer-multi-region?ref=<VERSION>"
}

# Include all settings from the root terragrunt.hcl file
include {
  path = find_in_parent_folders()
}

inputs = {
  iam_access_analyzer_type = "ACCOUNT"
  iam_access_analyzer_name = "iam-access-analyzer"
  seed_region              = "us-east-2"
}
----
```
If we went with my suggestion, then I think it would be good to remove any mention of the account baselines just to under complicate the mental model.

Reviewer 3: Ahh, I did have this initially, and then changed it back to a module, because I saw it in other examples. Let's stick to this in that case! I'm happy to change this first thing back on Monday!

Reviewer 3: Done. Apologies - with support and the monthly sync, this slipped my mind, but is done now. Thanks @robmorgan

Reviewer 2: Ok cool!

 uses IAM roles for EKS cluster workers.
* link:https://github.com/gruntwork-io/terraform-aws-ecs/tree/master/modules/ecs-cluster[`ecs-cluster`] creates IAM
 roles for ECS instances
* link:https://github.com/gruntwork-io/terraform-aws-lambda/tree/master/modules/lambda[`lambda`] creates IAM
 roles for Lambda functions

Use these modules whenever possible. You should always use IAM roles in your own modules any time you need to provide access to the AWS API. Using static API credentials should be avoided whenever possible.

[[cleanup_expired_certs_deployment]]
=== Automatically remove expired SSL/TLS certificates from IAM
Next, use the link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/blob/master/modules/cleanup-expired-certs/README.adoc[`cleanup-expired-certs`]
module to deploy a Lambda function to remove all expired SSL/TLS certificates from AWS IAM. This module will add all of the relevant IAM resources needed to
periodically execute the Lambda function and optionally, report metrics to CloudWatch. Be sure to deploy this module into each of your AWS accounts.

> If your using Terragrunt, then you can invoke the standalone module directly by following the pattern outlined in

Reviewer 4: ```suggestion
If you're using Terragrunt, then you can invoke the standalone module directly by following the pattern outlined in
```

<<wrapper_code_terragrunt>> and creating a `terragrunt.hcl` file with the following contents:

.infrastructure-live/<your account>/_global/cleanup-expired-certs/terragrunt.hcl
[source,hcl]
----
terraform {
  # Make sure to replace <VERSION> in this URL with the latest terraform-aws-cis-service-catalog release
  source = "git::ssh://git@github.com/gruntwork-io/terraform-aws-cis-service-catalog.git//modules/cleanup-expired-certs?ref=<VERSION>"
}

# Include all settings from the root terragrunt.hcl file
include {
  path = find_in_parent_folders()
}

inputs = {
  schedule_expression                = "rate(1 hour)"
  report_cloudwatch_metric           = true
  report_cloudwatch_metric_namespace = "custom/cis"
  report_cloudwatch_metric_name      = "cleanup-expired-iam-certs-count"
}
----

[[iam_access_analyzer_deployment]]
=== Steps to enable IAM Access Analyzer for all enabled AWS regions
To enable IAM Access Analyzer in your AWS Account or Organization, you need to do it separately for every region.

To enable this for your AWS account:

> * use the examples provided as part of `learning-and-testing/landingzone`:

Reviewer 4: NIT: This `*` is not rendering as a bullet point: https://deploy-preview-392--keen-clarke-470db9.netlify.app/guides/compliance/how-to-achieve-cis-benchmark-compliance/#iam_access_analyzer_deployment

Reviewer 4: Actually, this whole sentence should be removed, as CIS customers should NOT be using the `acccount-baseline` modules, as those don't take into account CIS requirements. Instead, they should solely use the `iam-access-analyzer` module as shown below.

Reviewer 3: ahaa! noted!

Reviewer 3: removed :) good shout!

link:https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-learning-and-testing/landingzone/account-baseline-root[account-baseline-root]
and link:https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-learning-and-testing/landingzone/account-baseline-security[account-baseline-security].

> If your using Terragrunt, then you can invoke the standalone module directly by following the pattern outlined in

Reviewer 4: ```suggestion
If you're using Terragrunt, then you can invoke the standalone module directly by following the pattern outlined in
```

<<wrapper_code_terragrunt>> and using a `terragrunt.hcl` file with the following contents:

.infrastructure-live/<your account>/_global/iam-access-analyzer/terragrunt.hcl
[source,hcl]
----
terraform {
# Make sure to replace <VERSION> in this URL with the latest terraform-aws-security release
source = "git::ssh://git@github.com/gruntwork-io/terraform-aws-security.git//modules/iam-access-analyzer-multi-region?ref=<VERSION>"
}

# Include all settings from the root terragrunt.hcl file
include {
  path = find_in_parent_folders()
}

inputs = {

> iam_access_analyzer_type = "ACCOUNT"
> iam_access_analyzer_name = "iam-access-analyzer"
> seed_region              = "us-east-2"

Reviewer 4: ```suggestion
  iam_access_analyzer_type = "ACCOUNT"
  iam_access_analyzer_name = "iam-access-analyzer"
  seed_region              = "us-east-2"
```

=== Use IAM roles for EC2 instances
All Gruntwork modules that require AWS API access use roles rather than an IAM user with static API credentials for
authentication. For example:

* link:https://github.com/gruntwork-io/terraform-aws-server/blob/master/modules/single-server/main.tf[`terraform-aws-server`]
 is used to manage a single EC2 instance with an IAM role attached.
* link:https://github.com/gruntwork-io/terraform-aws-asg[`terraform-aws-asg`] applies IAM roles to instances in auto-scaling
 group.
* link:https://github.com/gruntwork-io/terraform-aws-eks/blob/master/modules/eks-cluster-workers/main.tf[`terraform-aws-eks`]
 uses IAM roles for EKS cluster workers.
* link:https://github.com/gruntwork-io/terraform-aws-ecs/tree/master/modules/ecs-cluster[`ecs-cluster`] creates IAM
 roles for ECS instances
* link:https://github.com/gruntwork-io/terraform-aws-lambda/tree/master/modules/lambda[`lambda`] creates IAM
 roles for Lambda functions

Use these modules whenever possible. You should always use IAM roles in your own modules any time you need to provide access to the AWS API. Using static API credentials should be avoided whenever possible.

[[cleanup_expired_certs_deployment]]

> === Automatically remove expired SSL/TLS certificates from IAM

Reviewer 2: @brikis98 I could do something similar to the [Create VPC flow logs step](https://deploy-preview-392--keen-clarke-470db9.netlify.app/guides/compliance/how-to-achieve-cis-benchmark-compliance/#configure_logging), but that uses Terraform, _not_ Terragrunt examples. Alternatively, I could also reduce the amount of verbosity and follow something similar to the [Create IAM Users](https://deploy-preview-392--keen-clarke-470db9.netlify.app/guides/compliance/how-to-achieve-cis-benchmark-compliance/#iam_user_authentication) step. See a quick mockup below:

```
Next, use the link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/blob/master/modules/cleanup-expired-certs/README.adoc[`cleanup-expired-certs`]
module to deploy a Lambda function to remove all expired SSL/TLS certificates from AWS IAM. This module will add all of the relevant IAM resources needed to
periodically execute the Lambda function and optionally, report metrics to CloudWatch. Be sure to deploy this module into each of your AWS accounts.

Follow the pattern outlined in
<<wrapper_code>> to use the `cleanup-expired-certs` module, even though this isn't a wrapper module.

If you are using multiple AWS accounts, then be sure to deploy this module into each account.

If you are using Terraform then refer to the link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/examples/cleanup-expired-certs/terraform[example usage].
Alternatively, if you are using Terragrunt, refer to the following link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/examples/cleanup-expired-certs/terragrunt[example usage]
```

The main issue is there is a bit of a disparity in how we present each step.

> === Automatically remove expired SSL/TLS certificates from IAM

Reviewer 4: Hm, yea, the disparity is not great. I guess as part of the CIS Service Catalog project, we will need to update this guide to use the Service Catalog approach for everything.

In the meantime, perhaps we follow the [ebs-encryption example in Acme CIS](https://github.com/gruntwork-io/cis-infrastructure-live-acme/tree/master/dev/_global/ebs-encryption) and add that `terragrunt.hcl` here?

> === Automatically remove expired SSL/TLS certificates from IAM

Reviewer 3: example for [IAM Access Analyzer](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/392/commits/94b59520e055b357499cecb31e9ffe94c0f2a76c)

> === Automatically remove expired SSL/TLS certificates from IAM

Reviewer 2: Thanks everyone for the feedback, I've reworked it with the `cleanup-expired-certs` Terragrunt configuration.

Use these modules whenever possible. You should always use IAM roles in your own modules any time you need to provide access to the AWS API. Using static API credentials should be avoided whenever possible.

[[cleanup_expired_certs_deployment]]
=== Automatically remove expired SSL/TLS certificates from IAM

> Next, use the link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/blob/master/modules/cleanup-expired-certs/README.adoc[`cleanup-expired-certs`]

Reviewer 2: Note this link won't work until https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/pull/58 is merged as it contains the new documentation in AsciiDoc format.

Reviewer 2: Merged now, so resolving.

the use of root].  Fortunately, most of these activities are rare, so usage of the root account can be kept to
a minimum.

[[configure_storage]]
=== Configure storage

[[s3_buckets_deployment]]
==== S3 Buckets

> To make sure your S3 buckets are compliant with the benchmark, all you have to do is use the
> link:https://github.com/gruntwork-io/terraform-aws-security/tree/master/modules/private-s3-bucket[`private-s3-bucket` module].

Reviewer 4: ```suggestion
To make sure your S3 buckets are compliant with the benchmark, use the
link:https://github.com/gruntwork-io/terraform-aws-security/tree/master/modules/private-s3-bucket[`private-s3-bucket` module] to create and manage all of your S3 buckets. This module blocks public access and enforces encryption by default. Note that all Gruntwork modules that create S3 buckets use this module under the hood.
```

Reviewer 4: Oh, should we add a `terragrunt.hcl` example of using this module? Or better yet, the service in `aws-service-catalog` that wraps it?

Author: Fixed in https://github.com/gruntwork-io/gruntwork-io.github.io/pull/392/commits/5cd3f1c19af4b37e0f4ecd1980e2da0f1e7b2b00.

[[encrypt_ebs_volumes]]
==== Encrypt EBS volumes

Use the link:https://github.com/gruntwork-io/terraform-aws-security/tree/master/modules/ebs-encryption-multi-region[`ebs-encryption-multi-region` module]

> to configure AWS EBS encryption in all enabled regions of an AWS Account, as per the Benchmark recommendation.

Reviewer 4: Same: add a `terragrunt.hcl` example?

Reviewer 3: Added the example [here: ](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/392/commits/b1041b233908639b6ac7720ae690184e2e983520)

Reviewer 2: Just a heads up - I've reworked this for Terragrunt. I'll resolve.

the use of root].  Fortunately, most of these activities are rare, so usage of the root account can be kept to
a minimum.

[[configure_storage]]
=== Configure storage

[[s3_buckets_deployment]]
==== S3 Buckets
To make sure your S3 buckets are compliant with the benchmark, use the
link:https://github.com/gruntwork-io/terraform-aws-security/tree/master/modules/private-s3-bucket[`private-s3-bucket` module]
to create and manage all of your S3 buckets. This module blocks public access and enforces encryption by default. Note
that all Gruntwork modules that create S3 buckets use this module under the hood.

> To see a usage example of the `private-s3-bucket` module, refer the
> link:the https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/modules/data-stores/s3-bucket/[`s3-bucket` service]
> in the Gruntwork Service Catalog.

Reviewer 4: ```suggestion
You can either use the `private-s3-bucket` module in your own modules, or, if you wish to deploy a standalone S3 bucket, use the https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/modules/data-stores/s3-bucket/[`s3-bucket` service]
from the Gruntwork Service Catalog.
```

[[encrypt_ebs_volumes]]
==== Encrypt EBS volumes

Use the link:https://github.com/gruntwork-io/terraform-aws-security/tree/master/modules/ebs-encryption-multi-region[`ebs-encryption-multi-region` module]
to configure AWS EBS encryption in all enabled regions of an AWS Account, as per the Benchmark recommendation.

> If your using Terragrunt, then you can invoke the standalone module directly by following the pattern outlined in

Reviewer 4: ```suggestion
If you're using Terragrunt, then you can invoke the standalone module directly by following the pattern outlined in
```

<<wrapper_code_terragrunt>> and creating a `terragrunt.hcl` file with the following contents:

.infrastructure-live/<your account>/_global/ebs-encryption/terragrunt.hcl
[source,hcl]
----
terraform {
  # Make sure to replace <VERSION> in this URL with the latest terraform-aws-security release
  source = "git::ssh://git@github.com/gruntwork-io/terraform-aws-security.git//modules/ebs-encryption-multi-region?ref=<VERSION>"
}

# Include all settings from the root terragrunt.hcl file
include {
  path = find_in_parent_folders()
}

inputs = {
  enable_encryption = true
  seed_region       = "us-east-2"
}
----

[source,hcl]
----
# ----------------------------------------------------------------------------------------------------------------------

> # ELASTIC BLOCK STORAGE (EBS) ENCRYPTION DEFAULTS

Reviewer 4: What's this for? Seems like a left over example?

Reviewer 3: probs - will review and look how it looks now!

Reviewer 3: removed the whole block - it seems like we had what we needed in there.

1.18,<<iam_roles_for_instances>>,Use Gruntwork modules to ensure EC2 instances use roles for access
1.19,TODO(Rob),TODO
1.20,TODO(Ana),TODO

> 1.21,<<iam_access_analyzer>>,Use Gruntwork modules to enable IAM Access Analyzer across regions

Author: This does not render correctly - perhaps it needs to be `<<iam_access_analyzer_deployment>>`?

Reviewer 3: ahh of course - I renamed it!

[[configure_networking]]
=== Configure networking

If you're using Gruntwork's link:https://github.com/gruntwork-io/terraform-aws-vpc[VPC module] for your VPCs, two
of the four recommendations in this section are already taken care of! By default, none of our modules allow
security groups to access to ports 22 or 3389 from the world, and our architecture has a least-privileges-based routing
configuration by default.

To meet the 5.1 recommendation, you need to create your Network ACL Rules for the VPC-App using the
link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/modules/vpc-app-network-acls[`vpc-app-network-acls`
module] and your NACL Rules for the VPC-Mgmt (which is being deprecated) using the
link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/modules/vpc-mgmt-network-acls[`vpc-mgmt-network-acls`

> module].

Reviewer 4: Code examples plz!

Reviewer 2: @brikis98 @marinalimeira This one is a bit tricky... For the code examples, we need to:

1. Possibly link to this section: https://www.gruntwork.io/guides/compliance/how-to-achieve-cis-benchmark-compliance/#wrapper_modules
2. Explain how we extend the `vpc-app` and `vpc-mgmt` modules by creating new ones in `infrastructure-modules`.
3. Explain how in the new `vpc-app` and `vpc-mgmt` modules we need to invoke the `vpc-mgmt-network-acls` module.
4. Explain how to create a `terragrunt.hcl` file that invokes our new VPC modules in `infrastructure-modules`.

Is there an easier way to do this? Could we just do the last step?

Reviewer 4: I think the real solution is to add a VPC service to the CIS Service Catalog, which would include `vpc-app` and `vpc-app-network-acls`, and deploy that with one `terragrunt.hcl`. We have the ticket for this already: https://gruntwork.atlassian.net/browse/IAC-1643... But it's not part of the CIS v1.3 project.

So I guess it's a trade-off between:

1. Updating this guide with a bunch of steps that will take some time to put together and will be deprecated soon.
2. Doing the VPC service work now, and updating the guide to use that. But note that the VPC service would have to extend the existing VPC service in `terraform-aws-service-catalog`, and not all our CIS customers have access to that yet, so that would be yet another yak to shave.
3. Doing some minimal version of (1). That is, we say vaguely something like, "You should use the `vpc-app-network-acls` module to meet the new CIS v1.3 requirements. Check out our [VPC guide](https://gruntwork.io/guides/networking/how-to-deploy-production-grade-vpc-aws/) for how to deploy a VPC with the `vpc-app` module. Once that's deployed, add the `vpc-app-network-acls` module as follows: <snippet>. Deploy all this using Terragrunt: <snippet>."

It's not ideal, but (3) is probably the way to go, and we'll replace it fairly soon when the CIS Service Catalog work is done.

Reviewer 5: I am not sure if I am following... What is the difference between (2) and [the cis-infra-live-acme](https://github.com/gruntwork-io/cis-infrastructure-live-acme/tree/master/dev/us-east-2/dev/vpc) + cis-infra-modules?

Reviewer 4: The old way:

1. Customer has an `infra-modules` repo with a bunch of Terraform code in it.
2. Customers' `infra-live` repo sets the `source` URL to their own `infra-modules` repo.

The new way:

1. Gruntwork has an `service-catalog` repo with a bunch of Terraform code in it.
2. Customers' `infra-live` repo sets the `source` URL to Gruntwork's `service-catalog` repo.

The CIS Service Catalog project is moving us to the new way for CIS... But we're not there yet. So we're trying to decide if we want to do the work of showing customers the  "old way" for this VPC NACL stuff, only to throw it away a little later (option 1)... Or if we can go straight to the new way (option 2). Or some minimal version of the old way, as a placeholder until we finish the Service Catalog stuff (option 3).

Reviewer 3: I'd vouch with respect for time and simplicity, that we go with the old way. And then as part of CIS Service Catalog work, we come back here to update this.

Reviewer 5: I'll work on the `Updating this guide with a bunch of steps that will take some time to put together and will be deprecated soon.`!

1.5,<<root_mfa>>,Manually configure MFA for the root user
1.6,<<root_mfa>>,Use a Yubikey (or other hardware MFA) for the root user
1.7,<<manual_steps>>,Take manual steps to complete this recommendation
1.8-9,<<create_password_policy>>,Use the IAM password policy module
1.10,<<configure_authentication>>,Configure authentication using SAML or IAM
1.11,<<iam_user_authentication>>,Create IAM users with the `iam-users` module
1.12,<<next_steps>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused credentials
1.13,TODO,Use SecurityHub
1.14,<<next_steps>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused access keys
1.15,<<iam_user_authentication>>,Use IAM groups
1.16,<<iam_user_authentication>>,Use the Gruntwork modules to create best-practices groups and roles
1.17,<<iam_user_authentication>>,Use the `iam-groups` module to create a support group
1.18,<<iam_roles_for_instances>>,Use Gruntwork modules to ensure EC2 instances use roles for access
1.19,<<cleanup_expired_certs>>,Use Gruntwork modules to automatically remove expired certificates from IAM

> 1.20,TODO(Ana),TODO

Reviewer 2: @infraredgirl what did you want to say about the S3 stuff?

Author: Done - I've filled in the missing bits in the traceability matrix.
